### PR TITLE
Fixing tailing for events command

### LINKED ISSUE #565: govc events tailing has repeated events in the output
Sample output:
```
HMAHMOOD-MBPRO:Volumes hmahmood$ GOVC_URL=root:xxxxx@172.16.127.130 GOVC_INSECURE=1 govc events -f
[Mon Aug 22 10:27:46 2016] [event..category] Firewall configuration has changed. Operation 'add' for rule set cmmds succeeded.
[Mon Aug 22 10:27:46 2016] [event..category] Firewall configuration has changed. Operation 'enable' for rule set cmmds succeeded.
[Mon Aug 22 10:27:46 2016] [event..category] Firewall configuration has changed. Operation 'add' for rule set vsanvp succeeded.
...
[Mon Aug 22 10:30:01 2016] [info] User dcui@127.0.0.1 logged in as VMware-client/5.1.0
[Mon Aug 22 10:41:40 2016] [info] User root@127.0.0.1 logged in as VMware-client/5.1.0
[Mon Aug 22 10:41:41 2016] [info] User root@127.0.0.1 logged out (login time: Monday, 22 
...
[Mon Aug 22 10:27:46 2016] [event..category] Firewall configuration has changed. Operation 'enable' for rule set cmmds succeeded.
[Mon Aug 22 10:27:46 2016] [event..category] Firewall configuration has changed. Operation 'add' for rule set vsanvp succeeded.
[Mon Aug 22 10:27:46 2016] [event..category] Firewall configuration has changed. Operation 'enable' for rule set vsanvp succeeded.
[Mon Aug 22 10:27:46 2016] [event..category] Firewall configuration has changed. Operation 'noallowall' for rule set vsanvp succeeded.
[Mon Aug 22 10:27:46 2016] [event..category] Firewall configuration has changed. Operation 'setrequired' for rule set vsanvp succeeded.
[Mon Aug 22 10:27:46 2016] [event..category] Firewall configuration has changed. Operation 'add' for rule set rabbitmqproxy succeeded.
[Mon Aug 22 10:27:46 2016] [event..category] Firewall configuration has changed. Operation 'enable' for rule set rabbitmqproxy succeeded.
[Mon Aug 22 10:27:46 2016] [event..category] Firewall configuration has changed. Operation 'add' for rule set ipfam succeeded.
[Mon Aug 22 10:27:46 2016] [event..category] Firewall configuration has changed. Operation 'add' for rule set vvold succeeded.
[Mon Aug 22 10:27:46 2016] [event..category] Firewall configuration has changed. Operation 'add' for rule set iofiltervp succeeded.
[Mon Aug 22 10:27:46 2016] [event..category] Firewall configuration has changed. Operation 'add' for rule set esxupdate succeeded.
[Mon Aug 22 10:27:46 2016] [event..category] Firewall configuration has changed. Operation 'add' for rule set vit succeeded.
[Mon Aug 22 10:27:46 2016] [event..category] Firewall configuration has changed. Operation 'add' for rule set vsanhealth-multicasttest succeeded.
[Mon Aug 22 10:27:46 2016] [event..category] Port vmk0 is now protected by Firewall.
[Mon Aug 22 10:27:46 2016] [event..category] Network connectivity restored on virtual switch "vSwitch0", portgroups: "Management Network". Physical NIC vmnic0 is up.
[Mon Aug 22 10:27:46 2016] [event..category] At least one coredump target has been configured. Host core dumps will be saved.
[Mon Aug 22 10:27:46 2016] [event..category] The DCUI has been enabled.
[Mon Aug 22 10:27:46 2016] [event..category] SSH access has been enabled.
[Mon Aug 22 10:30:01 2016] [info] User dcui@127.0.0.1 logged in as VMware-client/5.1.0
[Mon Aug 22 10:41:40 2016] [info] User root@127.0.0.1 logged in as VMware-client/5.1.0
[Mon Aug 22 10:41:41 2016] [info] User root@127.0.0.1 logged out (login time: Monday, 22 August, 2016 15:41:40, number of API invocations: 0, user agent: VMware-client/5.1.0)
[Mon Aug 22 10:41:49 2016] [info] User root@127.0.0.1 logged in as VMware-client/5.1.0
[Mon Aug 22 10:41:49 2016] [info] User root@127.0.0.1 logged out (login time: Monday, 22 August, 2016 15:41:49, number of API invocations: 0, user agent: VMware-client/5.1.0)
[Mon Aug 22 10:48:18 2016] [info] User root@172.16.127.1 logged in as Go-http-client/1.1
[Mon Aug 22 10:48:49 2016] [info] User root@172.16.127.1 logged in as Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/52.0.2743.116 Safari/537.36
```

### PR DESCRIPTION
**Pull request by hmahmood**
Fixes #565

### CODE REVIEW

File: property/wait.go [hunk 1]
```diff
@@ -119,10 +120,12 @@ func WaitForView(ctx context.Context, c *Collector, view types.ManagedObjectRefe
 	if err != nil {
 		return err
 	}
-	return waitLoop(ctx, p, f)
+	return waitLoop(ctx, p, func(obj types.ManagedObjectReference, pc []types.PropertyChange) bool {
```
Comment (dougm) [inline property/wait.go]: Doesn't look like this change is needed, since 'f' has the same signature.

File: event/processor.go [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+	"github.com/vmware/govmomi/property"
+	"github.com/vmware/govmomi/view"
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+const latestPageProp = "latestPage"
+
+type tailInfo struct {
+	t              *eventTailer
+	obj, collector types.ManagedObjectReference
```
Comment (dougm) [inline event/processor.go]: style: prefer to stick with 1 field per line

File: event/processor.go [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+}
+
+type eventProcessor struct {
+	mgr      Manager
+	pageSize int32
+	tailers  map[string]*tailInfo // tailers by collector ref
```
Comment (dougm) [inline event/processor.go]: You can use types.ManagedObjectReference for the map key instead of string.

File: event/processor.go [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+		return false
+	})
+
+	return err
+}
+
+func (p *eventProcessor) process(c types.ManagedObjectReference, pc []types.PropertyChange) error {
+	t := p.tailers[c.String()]
+	if t == nil {
+		return fmt.Errorf("unknown collector %s", c.String())
```
Comment (dougm) [inline event/processor.go]: Can just use 'c' here, no need to explicitly call .String()

File: event/manager.go [hunk 5]
```diff
@@ -161,20 +161,16 @@ func (m Manager) EventCategory(ctx context.Context, event types.BaseEvent) (stri
 }
 
 // Get the events from the specified object(s) and optionanlly tail the event stream
-func (m Manager) Events(ctx context.Context, objects []types.ManagedObjectReference, pageSize int32, tail bool, force bool, f func([]types.BaseEvent) error) error {
+func (m Manager) Events(ctx context.Context, objects []types.ManagedObjectReference, pageSize int32, tail bool, force bool, f func(types.ManagedObjectReference, []types.BaseEvent) error) error {
```
Comment (dougm) [inline event/manager.go]: Maybe we should make 'EventProcessor' and exported type, with fields PageSize, Tail, Force, Callback, etc. ?  Would be a cleaner method signature that we can also extend in the future without breaking the method signature.

**Comment by dougm** [comment]
Looks good overall, just a few suggestions.

@cgtexmex can you also review?

File: event/manager.go (hunk 5 above)
Comment (hmahmood) [inline event/manager.go]: That makes sense, although I would call it something like `EventReader`.

Comment (hmahmood) [inline event/manager.go]: I am going to leave this out for now. Got some other issues in VIC that I need to deal with.

Comment (dougm) [inline event/manager.go]: ok, the type is named 'eventProcessor' in this PR, 'EventReader' is fine too - mainly concerned with the method signature.

**Pull request merged**